### ISSUE TITLE
[Phase 3] Morphology Classification

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/galaxy-morphology-photoz#4]
# Phase 3: Morphology Classification

**Parent Issue:** #1

## Objective

Train multi-class classifier for galaxy morphology using Galaxy Zoo labels.

## Dependencies

- Phase 1 completed (unified catalog)
- Phase 2 completed (features extracted)

### Label Processing
- [ ] Convert Galaxy Zoo vote fractions to class labels
- [ ] Implement majority vote (>50% agreement)
- [ ] Analyze class distribution and imbalance
- [ ] Define morphology classes: Spiral, Elliptical, Irregular/Merger

### Model Implementation
- [ ] Implement `src/models/morphology/classifier.py`
- [ ] Create XGBoost multi-class wrapper
- [ ] Add hyperparameter configuration via YAML
- [ ] Implement model saving/loading

### Training Pipeline
- [ ] Implement `src/models/morphology/train.py`
- [ ] Handle class imbalance (sample weights or SMOTE)
- [ ] Implement stratified k-fold cross-validation
- [ ] Add experiment logging

### Evaluation
- [ ] Implement `src/models/morphology/evaluate.py`
- [ ] Compute accuracy, precision, recall per class
- [ ] Generate multi-class confusion matrix
- [ ] Analyze misclassifications (ambiguous morphology?)

### Label Extraction
```python
def get_morphology_label(gz_row: pd.Series) -> str:
    """Extract morphology class from Galaxy Zoo vote fractions.

    Args:
        gz_row: Row with vote fraction columns

    Returns:
        'spiral', 'elliptical', or 'irregular'
    """
    # GZ2 columns: smooth_or_featured, has_spiral_arms, etc.
    if gz_row['Class1.1'] > 0.5:  # Smooth
        return 'elliptical'
    elif gz_row['Class1.2'] > 0.5:  # Features/disk
        if gz_row['Class2.1'] > 0.5:  # Has spiral arms
            return 'spiral'
    return 'irregular'
```

### XGBoost Classifier
```python
from xgboost import XGBClassifier

class MorphologyClassifier:
    def __init__(self, n_estimators: int = 200):
        self.model = XGBClassifier(
            n_estimators=n_estimators,
            max_depth=6,
            learning_rate=0.1,
            objective='multi:softmax',
            num_class=3,
            use_label_encoder=False,
            eval_metric='mlogloss'
        )
        self.label_encoder = LabelEncoder()

    def fit(self, X: np.ndarray, y: np.ndarray):
        y_encoded = self.label_encoder.fit_transform(y)
        self.model.fit(X, y_encoded)

    def predict(self, X: np.ndarray) -> np.ndarray:
        y_pred = self.model.predict(X)
        return self.label_encoder.inverse_transform(y_pred)
```

## Verification

```bash
python src/models/morphology/train.py --output models/morphology_xgb.pkl
python src/models/morphology/evaluate.py --model models/morphology_xgb.pkl
# Expected: Accuracy: 0.87, Per-class F1 scores
```

## Definition of Done

- [ ] Classifier achieves >85% accuracy on test set
- [ ] Per-class precision/recall documented
- [ ] Confusion matrix saved
- [ ] Top 10 important features identified
- [ ] Model checkpoint saved and loadable
